### PR TITLE
upd

### PR DESCRIPTION
**Pull request by idcooldi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,74 +48,4 @@ func main() {
 	updates, err := bot.GetUpdatesChan(u)
 
 	for update := range updates {
-		if update.Message == nil { // ignore any non-Message Updates
```
Comment (OGKevin) [inline README.md]: why did you delete all this info from the README?

File: bot.go [hunk 2]
```diff
@@ -33,10 +34,31 @@ type BotAPI struct {
 // NewBotAPI creates a new BotAPI instance.
 //
 // It requires a token, provided by @BotFather on Telegram.
-func NewBotAPI(token string) (*BotAPI, error) {
+func NewBotAPI(token string, proxyparam ...interface{}) (*BotAPI, error) {
```
Comment (OGKevin) [inline bot.go]: instead of passing `...interface{}` here, i would suggest you create a config struct that contains the data that needs to be passed.

File: bot.go [hunk 3]
```diff
@@ -33,10 +34,31 @@ type BotAPI struct {
...
+ 		var auth *proxy.Auth
+		if proxyparam[0] == "socks5" {
+			if proxyparam[2] == nil {
+				auth = nil
+			} else {
+				auth = proxyparam[2].(*proxy.Auth)
+			}
+			dialer, err := proxy.SOCKS5("tcp", proxyparam[1].(string), auth, proxy.Direct)
+			if err != nil {
+				return nil, err
```
Comment (OGKevin) [inline bot.go]: I would suggest using `errors.Wrap(err, "telegram: explanation")` here. So that the returned error has more context.

**Comment by jqs7** [comment]
why not `tgbotapi.NewBotAPIWithClient`?

BTW, by default, tgbotapi uses default http client that uses `http.DefaultTransport` and it support `ProxyFromEnvironment`, so you could just set `https_proxy=socks5://url-proxy` environment variable to make everything pass through proxy.

hope my suggestion helpful. 😃

**Comment by Syfaro** [comment]
I'm not sure why people keep making PRs to add proxy configuration into this library. Not only is it completely possible to do out of library, but there have been 4 other PRs attempting to do the same thing which have all been closed with a message saying it's out of scope.
